[FSDP2] Added `_to_kwargs` root forward input cast

awgu opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #118584
* #118298
* #118755
* #118223
* #118136
* #118118
* #118017
* #118004
* #117975
* #117973
* __->__ #117955
* #117950

This PR adds a `_to_kwargs()` call on the FSDP root module's forward inputs to move them to `device` similar to DDP.
https://github.com/pytorch/pytorch/blob/39df084001c54cca5fe3174176f9b0206ddb7dcf/torch/nn/parallel/distributed.py#L1426-L1427

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225

Commit:
[FSDP2] Added `_to_kwargs` root forward input cast

[ghstack-poisoned]

Commit:
Update on "[FSDP2] Added `_to_kwargs` root forward input cast"




cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225

[ghstack-poisoned]

Commit:
Update on "[FSDP2] Added `_to_kwargs` root forward input cast"


This PR adds a `_to_kwargs()` call on the FSDP root module's forward inputs to move them to `device` similar to DDP.
https://github.com/pytorch/pytorch/blob/39df084001c54cca5fe3174176f9b0206ddb7dcf/torch/nn/parallel/distributed.py#L1426-L1427

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225

[ghstack-poisoned]

Skylion007 commented on torch/distributed/_composable/fsdp/_fsdp_state.py:
```diff
@@ -31,6 +34,13 @@ def _root_pre_forward(
+                    args_tuple, kwargs_tuple = _to_kwargs(
+                        args, kwargs, self._device, False
+                    )  # same as DDP
+                args, kwargs = (args_tuple[0], kwargs_tuple[0])
```
```suggestion
                args, kwargs = args_tuple[0], kwargs_tuple[0]
```

Commit:
Update on "[FSDP2] Added `_to_kwargs` root forward input cast"


This PR adds a `_to_kwargs()` call on the FSDP root module's forward inputs to move them to `device` similar to DDP.
https://github.com/pytorch/pytorch/blob/39df084001c54cca5fe3174176f9b0206ddb7dcf/torch/nn/parallel/distributed.py#L1426-L1427

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225

[ghstack-poisoned]

Commit:
Update on "[FSDP2] Added `_to_kwargs` root forward input cast"


This PR adds a `_to_kwargs()` call on the FSDP root module's forward inputs to move them to `device` similar to DDP.
https://github.com/pytorch/pytorch/blob/39df084001c54cca5fe3174176f9b0206ddb7dcf/torch/nn/parallel/distributed.py#L1426-L1427

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225

[ghstack-poisoned]

Commit:
Update on "[FSDP2] Added `_to_kwargs` root forward input cast"


This PR adds a `_to_kwargs()` call on the FSDP root module's forward inputs to move them to `device` similar to DDP.
https://github.com/pytorch/pytorch/blob/39df084001c54cca5fe3174176f9b0206ddb7dcf/torch/nn/parallel/distributed.py#L1426-L1427

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225

[ghstack-poisoned]

weifengpy commented on torch/distributed/_composable/fsdp/_fsdp_state.py:
```diff
@@ -71,6 +84,16 @@ def _init_fqns(self) -> None:
                 module_to_fsdp_param_group[module]._module_fqn = module_name
 
+<<<<<<< HEAD
+=======
```
will stamp once resolving this merging conflict

awgu replied:
Oops 😅  I fixed it now.

Commit:
Update on "[FSDP2] Added `_to_kwargs` root forward input cast"


This PR adds a `_to_kwargs()` call on the FSDP root module's forward inputs to move them to `device` similar to DDP.
https://github.com/pytorch/pytorch/blob/39df084001c54cca5fe3174176f9b0206ddb7dcf/torch/nn/parallel/distributed.py#L1426-L1427

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225

[ghstack-poisoned]

Commit:
Update on "[FSDP2] Added `_to_kwargs` root forward input cast"


This PR adds a `_to_kwargs()` call on the FSDP root module's forward inputs to move them to `device` similar to DDP.
https://github.com/pytorch/pytorch/blob/39df084001c54cca5fe3174176f9b0206ddb7dcf/torch/nn/parallel/distributed.py#L1426-L1427

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225

[ghstack-poisoned]

Commit:
Update on "[FSDP2] Added `_to_kwargs` root forward input cast"


This PR adds a `_to_kwargs()` call on the FSDP root module's forward inputs to move them to `device` similar to DDP.
https://github.com/pytorch/pytorch/blob/39df084001c54cca5fe3174176f9b0206ddb7dcf/torch/nn/parallel/distributed.py#L1426-L1427

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225

[ghstack-poisoned]

Commit:
Update on "[FSDP2] Added `_to_kwargs` root forward input cast"


This PR adds a `_to_kwargs()` call on the FSDP root module's forward inputs to move them to `device` similar to DDP.
https://github.com/pytorch/pytorch/blob/39df084001c54cca5fe3174176f9b0206ddb7dcf/torch/nn/parallel/distributed.py#L1426-L1427

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225

[ghstack-poisoned]

Commit:
Update on "[FSDP2] Added `_to_kwargs` root forward input cast"


This PR adds a `_to_kwargs()` call on the FSDP root module's forward inputs to move them to `device` similar to DDP.
https://github.com/pytorch/pytorch/blob/39df084001c54cca5fe3174176f9b0206ddb7dcf/torch/nn/parallel/distributed.py#L1426-L1427

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225

[ghstack-poisoned]

Commit:
Update on "[FSDP2] Added `_to_kwargs` root forward input cast"


This PR adds a `_to_kwargs()` call on the FSDP root module's forward inputs to move them to `device` similar to DDP.
https://github.com/pytorch/pytorch/blob/39df084001c54cca5fe3174176f9b0206ddb7dcf/torch/nn/parallel/distributed.py#L1426-L1427

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225

[ghstack-poisoned]

Commit:
Update on "[FSDP2] Added `_to_kwargs` root forward input cast"


This PR adds a `_to_kwargs()` call on the FSDP root module's forward inputs to move them to `device` similar to DDP.
https://github.com/pytorch/pytorch/blob/39df084001c54cca5fe3174176f9b0206ddb7dcf/torch/nn/parallel/distributed.py#L1426-L1427

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225

[ghstack-poisoned]

Commit:
Update on "[FSDP2] Added `_to_kwargs` root forward input cast"


This PR adds a `_to_kwargs()` call on the FSDP root module's forward inputs to move them to `device` similar to DDP.
https://github.com/pytorch/pytorch/blob/39df084001c54cca5fe3174176f9b0206ddb7dcf/torch/nn/parallel/distributed.py#L1426-L1427

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225

[ghstack-poisoned]

Commit:
Update on "[FSDP2] Added `_to_kwargs` root forward input cast"


This PR adds a `_to_kwargs()` call on the FSDP root module's forward inputs to move them to `device` similar to DDP.
https://github.com/pytorch/pytorch/blob/39df084001c54cca5fe3174176f9b0206ddb7dcf/torch/nn/parallel/distributed.py#L1426-L1427

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225

[ghstack-poisoned]

Commit:
Update on "[FSDP2] Added `_to_kwargs` root forward input cast"


This PR adds a `_to_kwargs()` call on the FSDP root module's forward inputs to move them to `device` similar to DDP.
https://github.com/pytorch/pytorch/blob/39df084001c54cca5fe3174176f9b0206ddb7dcf/torch/nn/parallel/distributed.py#L1426-L1427

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225

[ghstack-poisoned]

awgu commented:
@pytorchbot merge
